src/core/interpreter.cc

```diff
@@ -563,18 +563,25 @@ int RedisLogCommand(lua_State* lua) {
 // See https://www.lua.org/manual/5.3/manual.html#lua_Alloc
 void* mimalloc_glue(void* ud, void* ptr, size_t osize, size_t nsize) {
   (void)ud;
+  auto& stats = InterpreterManager::tl_stats();
   if (nsize == 0) {
-    InterpreterManager::tl_stats().used_bytes -= mi_usable_size(ptr);
+    const auto usable_size = mi_usable_size(ptr);
+    LOG_IF(WARNING, stats.used_bytes < usable_size)
```

Comment:
please always use LOG_EVERY_T for such warnings otherwise you will flood the logs, it's high frequency code.